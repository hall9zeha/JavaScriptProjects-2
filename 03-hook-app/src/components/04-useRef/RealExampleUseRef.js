import React, { useState } from 'react'
import { MultipleCustomHooks } from '../03-Examples/MultipleCustomHooks'

export default function RealExampleUseRef() {

    const [show, setShow] = useState(false)

  return (
    <div>
        <h1>RealExampleUseRef</h1>
        <hr/>
        { show && <MultipleCustomHooks data-testid='multiple-custom-hooks'/>}
        <button className='btn btn-primary mt-3'
                onClick={()=>setShow(!show)}
        >
            { show ? "Ocultar" : "Mostrar" }
        </button>
    </div>
  )
}
