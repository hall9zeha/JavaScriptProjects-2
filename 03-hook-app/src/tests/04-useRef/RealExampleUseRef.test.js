import {render, renderHook, screen, waitFor, act, fireEvent} from '@testing-library/react';

import renderer from "react-test-renderer";
import RealExampleUseRef from '../../components/04-useRef/RealExampleUseRef';

describe('Tests in RealExampleUseRef', () => {

    test('should show successful',()=>{
        const wrapper = renderer.create(<RealExampleUseRef/>).toJSON();
        expect(wrapper).toMatchSnapshot();

    });

    test('should show the <MultipleCustomHooks/> component',()=>{
        
        render(<RealExampleUseRef/>);
        
        const btnShowComponent = screen.getByRole('button',{name:'Mostrar'});
        const childComponetHide = screen.queryByTestId('multiple-custom-hooks');      
        
        expect(childComponetHide).toBeNull()//La primera vez no debe mostrarse
        fireEvent.click(btnShowComponent);//Al hacer click en mostrar

        const childComponetShow = screen.queryByTestId('multiple-custom-hooks');
        
        expect(childComponetShow).toBeTruthy()//Debe mostrarse


        
    });
 })