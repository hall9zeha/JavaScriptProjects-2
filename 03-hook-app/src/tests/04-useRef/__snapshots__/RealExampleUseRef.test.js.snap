// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in RealExampleUseRef should show successful 1`] = `
<div>
  <h1>
    RealExampleUseRef
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-3"
    onClick={[Function]}
  >
    Mostrar
  </button>
</div>
`;
